src/plugins/intel_gpu/tests/test_cases/prior_box_gpu_test.cpp

```diff
@@ -0,0 +1,106 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#include <algorithm>
+#include <intel_gpu/primitives/data.hpp>
+#include <intel_gpu/primitives/input_layout.hpp>
+#include <intel_gpu/primitives/prior_box.hpp>
+#include <random>
+#include <vector>
+
+#include "test_utils.h"
+
+using namespace cldnn;
+using namespace ::tests;
+
+namespace {
+
+template <class OutputType, class InputType>
+using prior_box_param = std::tuple<format,                       // Input and output format
+                                   std::vector<InputType>,       // output_size
+                                   std::vector<InputType>,       // image_size
+                                   cldnn::prior_box_attributes,  // attributes
+                                   std::vector<OutputType>>;     // expected values
+
+template <class OutputType, class InputType>
+class PriorBoxGPUTest : public ::testing::TestWithParam<prior_box_param<OutputType, InputType>> {
+public:
+    void SetUp() override {
+        format fmt{format::bfyx};
+        std::vector<InputType> output_size{};
+        std::vector<InputType> image_size{};
+        cldnn::prior_box_attributes attributes;
+        std::vector<OutputType> expected_values;
+
+        std::tie(fmt, output_size, image_size, attributes, expected_values) = this->GetParam();
+
+        memory::ptr output_size_layout = engine_.allocate_memory({type_to_data_type<InputType>::value, fmt, tensor{2}});
+        set_values<InputType>(output_size_layout, output_size);
+        memory::ptr image_size_layout = engine_.allocate_memory({type_to_data_type<InputType>::value, fmt, tensor{2}});
+        set_values<InputType>(image_size_layout, image_size);
+
+        topology tp;
+        tp.add(data("output_size", output_size_layout));
+        tp.add(data("image_size", image_size_layout));
+
+        std::vector<primitive_id> inputs{"output_size", "image_size"};
+        auto prior_box = cldnn::prior_box("prior_box",
+                                          inputs,
+                                          static_cast<int32_t>(output_size[0]),
+                                          static_cast<int32_t>(output_size[1]),
+                                          static_cast<int32_t>(image_size[0]),
+                                          static_cast<int32_t>(image_size[1]),
+                                          attributes,
+                                          type_to_data_type<OutputType>::value);
+        tp.add(prior_box);
+
+        network network(engine_, tp);
+
+        auto outputs = network.execute();
+
+        EXPECT_EQ(outputs.begin()->first, "prior_box");
```

Comment:
I think this check is a bit excessive - we're testing prior_box primitive here, not cldnn::network::execute() method.